### PR TITLE
Added demo annotation support.

### PR DESCRIPTION
**Pull request by usergenic**
- Jsdocs with `@demo` annotations now add entries to the `demos` collection for Element, ElementMixin, PolymerElement and Behavior.
 - [x] CHANGELOG.md has been updated
 - This addresses https://github.com/Polymer/polymer-analyzer/issues/578 but does not include `@demo` tag support in HTMLComments yet.

### CODE REVIEW

File: src/polymer/polymer-element.ts [hunk 1]
```diff
@@ -179,6 +179,7 @@ export class ScannedPolymerElement extends ScannedElement implements
   }
```
Comment (rictic) [inline src/polymer/polymer-element.ts]: Update to clang-format.48 to match this repo's updated clang-format and re-run `npm run format`?

File: src/model/element-base.ts [hunk 2]
```diff
@@ -41,6 +41,19 @@ export abstract class ScannedElementBase implements Resolvable {
...
 
+  applyJsdocDemoTags(baseUrl: string): void {
+    if (!this.jsdoc || !this.jsdoc.tags) {
+      return;
+    }
+    this.jsdoc.tags.filter((tag) => tag.tag === 'demo' && tag.name)
+        .forEach((tag) => {
+          this.demos.push({
+            desc: tag.description || undefined,
+            path: tag.name ? url.resolve(baseUrl, tag.name) : ''
```
Comment (rictic) [inline src/model/element-base.ts]: no need to switch on tag.name, as you added it to `filter`?

File: src/model/element-base.ts [hunk 3]
```diff
@@ -107,4 +120,4 @@ export abstract class ElementBase implements Feature {
   emitEventMetadata(_event: Event): Object {
     return {};
   }
-}
+}
```
Comment (rictic) [inline src/model/element-base.ts]: newline at end of file. recommend configuring your editor to automatically add this

File: src/polymer/polymer-element.ts (hunk 1 above)
Comment (usergenic) [inline src/polymer/polymer-element.ts]: Reformatted; done.

File: src/model/element-base.ts (hunk 2 above)
Comment (usergenic) [inline src/model/element-base.ts]: Good point!  Removed the ternary.

File: src/model/element-base.ts (hunk 3 above)
Comment (usergenic) [inline src/model/element-base.ts]: Done.  Also, updated vscode settings.json for project to do that https://github.com/Polymer/polymer-analyzer/pull/599

**Comment by usergenic** [comment]
Feedback addressed @rictic PTAL Thanks!

**Pull request merged**